main.js

```diff
@@ -218,3 +218,27 @@ document.querySelector('#form').addEventListener('submit', (e) => {
 
 /* Contact Form Validation end */
 
+
+/* preserve data in the browser start*/
+
+let userDetails = {
+  name: '',
+  email: '',
+  message: '',
+};
+
+if (localStorage.getItem('savedDetails') !== null) {
+  const finalDetails = localStorage.getItem('savedDetails');
+  userDetails = JSON.parse(finalDetails);
+}
+const input = document.querySelectorAll('input,textarea');
```

Comment:
- [ ] [OPTIONAL] I believe, it would be better if you add a class to the `input` fields that you want to select and use `querySelector()` with that class. Because maybe in the future you want to add another new input field somewhere in the website. This way your code is going to be scalable.